unifiednlp-base/src/main/java/org/microg/tools/selfcheck/NlpOsCompatChecks.java

```diff
@@ -47,7 +48,7 @@ public void doChecks(Context context, ResultCollector collector) {
     }
 
     private boolean checkSystemIsSupported(Context context, ResultCollector collector) {
-        boolean isSupported = (SDK_INT >= KITKAT && SDK_INT <= M);
+        boolean isSupported = (SDK_INT >= KITKAT && SDK_INT <= N_MR1);
```

Comment:
To be precise: UnifiedNlp does not work on AOSP Android 7.1 and some custom ROMs, including original LineageOS. Some LineageOS-based mods as well as some other custom ROMs do support UnifiedNlp. With support I mean: installed as a usual application. When UnifiedNlp is installed to `/system` it works on the aforementioned systems. AFAIK there is no proper way to find out if the system does support non-`/system` UnifiedNlp.
I think updating the check to accurately state that the system *might be unsupported* is the better way to go.

Reply:
Well, as I've created specific flavor for intent based version, I can change the check according flavor. What's your opinion?

Reply:
There are some reasons why I don't like having an additional flavor:
- It would be great to have the intent-based system working no matter which UnifiedNlp flavor is installed.
- The `UnifiedNlp` flavor was actually intended for this kind of usage (although the original network location service api is probably to complex to be implemented client side)
- There are already three flavors and quiet some users tend to take the wrong one already.
- We are overpolluting F-Droid with all this flavors ;) - and specifically: your current implementation uses the same package name as the `UnifiedNlp` flavor, which is not supported by F-Droid (two different packages with the same package name).

Reply:
I completely agree with you about flavors and I tought it's possible. But I've found today that Android on my phone starts to reject location updates with SecurityException and complains with android.permission.INSTALL_LOCATION_PROVIDER permission.

Reply:
**My opinion:**
Wouldn't be better to just add the changes to all flavors?
So, if UnifiedNlp bind to the system then it can be queried with both standard and custom way instead if the binding fail then it can be queried only with the custom way.

So apps that use the custom way do works in all cases.